### PR TITLE
feat: my orders/my tickets widget migration from summit registration

### PR DESCRIPTION
**Pull request by jaredhill4**
**What kind of change does this PR introduce?**

This PR introduces the new my orders/my tickets feature, per this ticket in Avaza:

https://tipit.avaza.com/task#!sortby=DateUpdatedDesc&task=2908629

It includes:

* A new page for the feature
* A widget-specific directory in the `components` directory
* store, actions, reducers (mostly copied from the `summit-registration` repo, but updated to meet the needs for this feature)
* components and sub-components for the order and ticket lists, as well as various popups needed to achieve full functionality.

**Does this PR introduce a breaking change?**

There are no known breaking changes introduced by these updates.

**What needs to be documented once your changes are merged?**

### CODE REVIEW

File: src/components/MyOrdersTicketsComponent.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+export const MyOrdersTicketsComponent = () => {
+    const user = useSelector(state => state.userState);
+    const summit = useSelector(state => state.summitState.summit);
+
+    if (!summit) return null;
+
+    const widgetProps = {
+        apiBaseUrl: getEnvVariable(SUMMIT_API_BASE_URL),
+        clientId: getEnvVariable(OAUTH2_CLIENT_ID),
```
Comment (smarcet) [inline src/components/MyOrdersTicketsComponent.js:16]: is this used for something ?

File: src/components/summit-my-orders-tickets/components/OrderList/OrderList.js [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+        <>
+            <h2 className="order-list-title">My Purchase Orders</h2>
+
+            {/* TODO: Replace with inline `Loading` component. */}
+            {(isLoading) && (
+                <div className="order-list-loading">{t("orders.loading")}</div>
+            )}
+
+            {/* TODO: Replace with `Empty` component. */}
+            {(!isLoading && (!hasOrders || !hasSummits)) && (
```
Comment (smarcet) [inline src/components/summit-my-orders-tickets/components/OrderList/OrderList.js:50]: what is the purpose of the hasSummits ? we have only one summit

File: src/components/summit-my-orders-tickets/store/reducers/base-reducer.js [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+import { START_LOADING, STOP_LOADING, LOGOUT_USER, RECEIVE_COUNTRIES } from "openstack-uicore-foundation/lib/utils/actions";
+import { RECEIVE_MARKETING_SETTINGS, CLEAR_MARKETING_SETTINGS } from '../actions/summit-actions';
+
+const DEFAULT_STATE = {
+    loading: false,
+    countries: [],
```
Comment (smarcet) [inline src/components/summit-my-orders-tickets/store/reducers/base-reducer.js:19]: this is not used anymore

File: src/components/summit-my-orders-tickets/store/reducers/user-reducer.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+const DEFAULT_STATE = {};
+
+const userReducer = (state = DEFAULT_STATE, action) => {
```
Comment (smarcet) [inline src/components/summit-my-orders-tickets/store/reducers/user-reducer.js:18]: do u need this reducer ?

File: src/components/summit-my-orders-tickets/components/OrderList/OrderListItem.js [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+import { useSelector } from 'react-redux';
+import classNames from 'classnames';
+import { OrderDetails } from '../OrderDetails/OrderDetails';
+import { OrderSummary } from '../OrderSummary/OrderSummary';
+import { OrderTicketList } from '../OrderTicketList/OrderTicketList';
+import { OrderOptions } from "../OrderOptions/OrderOptions";
+import { useOrderListContext } from "./OrderList.helpers";
+
+export const OrderListItem = ({ order, className }) => {
+    const summit = useSelector(state => state.summitState.summits.find(summit => summit.id == order.summit_id));
```
Comment (smarcet) [inline src/components/summit-my-orders-tickets/components/OrderList/OrderListItem.js:11]: why doing this if u already have the current summit as precondition
just pass the current summit as prop

File: src/components/summit-my-orders-tickets/store/reducers/base-reducer.js (hunk 3 above)
Comment (jaredhill4) [reply src/components/summit-my-orders-tickets/store/reducers/base-reducer.js]: Done!

File: src/components/summit-my-orders-tickets/components/OrderList/OrderListItem.js (hunk 5 above)
Comment (jaredhill4) [reply src/components/summit-my-orders-tickets/components/OrderList/OrderListItem.js]: Done!

File: src/components/summit-my-orders-tickets/components/OrderList/OrderList.js [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+                <div className="order-list-empty">{t("orders.empty")}</div>
+            )}
+
+            {(hasOrders && hasSummits) && (
```
[the remark this answers is not in the document]
Comment (jaredhill4) [reply src/components/summit-my-orders-tickets/components/OrderList/OrderList.js]: Done!

File: src/components/summit-my-orders-tickets/store/reducers/summit-reducer.js [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+} from "../actions/summit-actions";
+
+
+const DEFAULT_STATE = {
+    loading: true,
+    purchaseSummit: {},
+    selectedSummit: {
+        refund_policy: null,
+    },
+    summits: [],
```
[the remark this answers is not in the document]
Comment (jaredhill4) [reply src/components/summit-my-orders-tickets/store/reducers/summit-reducer.js]: Done!

**Review by smarcet — APPROVED** [review]
LGTM

**Pull request merged**